### PR TITLE
fix(cairo): various failures when column sizes are near page size

### PR DESCRIPTION
**Pull request by ideoma**
- fixed a failure in column resizing on writer close when var column size is near mapping page size.
- fixed a failure in commit with lag when var column size in lag exceeds o3 memory allocation page size.
- fixed SIGBUS in shift_copy when var column with column top is o3 merged

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

File: core/src/main/java/io/questdb/cairo/TableWriter.java [hunk 1]
```diff
@@ -4612,8 +4612,11 @@ private void setColumnSize(int columnIndex, long size, boolean doubleAllocate) {
                     if (doubleAllocate) {
                         mem2.allocate(pos * Long.BYTES + Long.BYTES);
                     }
-                    mem2.jumpTo(pos * Long.BYTES + Long.BYTES);
-                    m1pos = Unsafe.getUnsafe().getLong(mem2.getAppendAddress() - 8);
+                    // Jump to the number of records written to read length of var column correctly
+                    mem2.jumpTo(pos * Long.BYTES);
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/TableWriter.java:4616]: the intent behind is to ensure side, "under-jumping" could mean there is no 8 bytes to read after that. The only reason it works is that page size uniquely divisibly by 8 and this doesn't happen or very hard to make happen. Nonetheless logically this is incorrect

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 0 / 18 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|0|4|00.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|0|5|00.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 0 / 18 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|0|4|00.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|0|3|00.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|0|5|00.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 18 / 18 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Review by puzpuzpuz — DISMISSED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 31 / 31 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 31 / 31 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 31 / 31 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 33 / 33 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|18|18|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 31 / 31 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 31 / 31 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableReaderSelectedColumnRecord.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3CopyJob.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/O3OpenColumnJob.java|16|16|100.00%|
|:large_blue_circle:|io/questdb/cairo/vm/MemoryCARWImpl.java|5|5|100.00%|

**Review by bluestreak01 — APPROVED** [review]

**Pull request merged**